### PR TITLE
improve environment variable inheritance

### PR DESCRIPTION
**Pull request by berrydenhartog**
## Description

Load .env by default for test_engine_app but allow overwriting the values with environment variables

## Motivation and Context

To allow flexibility in how to run the applications some changes are preferred in how the variables are loaded. 

Also added 127.0.0.1 to ALLOW_ORIGINS since some tools prefer using this. (VSCode) 

## How to Test

run from test-engine-app: pytest tests/test_environment_variables.py

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [X]  I have tested the changes locally and verified that they work as expected.
- [ ]  I have added or updated the necessary documentation (README, API docs, etc.).
- [X]  I have added appropriate unit tests or functional tests for the changes made.
- [X]  I have followed the project's coding conventions and style guidelines.
- [X]  I have rebased my branch onto the latest commit of the main branch.
- [X]  I have squashed or reorganized my commits into logical units.
- [x]  I have added any necessary dependencies or packages to the project's build configuration.
- [X]  I have performed a self-review of my own code.
- [X]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

File: test-engine-app/test_engine_app/config/environment_variables.py [hunk 1]
```diff
@@ -20,46 +20,43 @@ class EnvironmentVariables:
...
-                    EnvironmentVariables._api_server_port = api_server_port
-                else:
-                    # Validation failed. Use defaults.
-                    pass
+                EnvironmentVariables._redis_consumer_group = consumer_group
+                EnvironmentVariables._redis_server_hostname = hostname
+                EnvironmentVariables._redis_server_port = server_port
+                EnvironmentVariables._api_server_port = api_server_port
+            else:
+                # Validation failed. Use defaults.
```
[the remark this answers is not in the document]
Comment (berrydenhartog) [reply test-engine-app/test_engine_app/config/environment_variables.py:58]: I do not think so. Default behavior seems to be to pickup the default values from the module

File: test-engine-app/test_engine_app/config/environment_variables.py [hunk 2]
```diff
@@ -20,46 +20,43 @@ class EnvironmentVariables:
...
+            error_count, _ = EnvironmentVariables._validate_data(
+                core_modules_folder,
+                validation_schemas_folder,
+                consumer_group,
+                hostname,
+                server_port,
+                api_server_port,
+            )
+
+            if error_count == 0:
```
[the remark this answers is not in the document]
Comment (berrydenhartog) [reply test-engine-app/test_engine_app/config/environment_variables.py:48]: The original code was written as this, tried to make minimal changes to the original source

File: test-engine-app/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import os
+
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def test_setup_and_teardown():
+    old_environ = dict(os.environ)
+    print("berry", old_environ)
```
Comment (anneschuth) [inline test-engine-app/tests/conftest.py:9]: Do we need to print `Berry`? :)

Comment (berrydenhartog) [reply test-engine-app/tests/conftest.py:9]: oops

Comment (berrydenhartog) [reply test-engine-app/tests/conftest.py]: fixed

**Comment by kimeetok** [comment]
Thanks @berrydenhartog for this!  
Just to share on the progress of this, our devs @imda-lionelteo @imda-kelvinkok @imda-amdlahir @imda-peckyoke have tested/ reviewed this together with #258.
We are good to go once @imda-kwokwk @imda-benedictlee have validated integration and deployments are ok :)

**Comment by imda-benedictlee** [comment]
Hi @anneschuth, my apologies, I need your help to pull changes from main into this branch as main contains Integration Test GitHub Action workflows that have been modified recently.

**Comment by berrydenhartog** [comment]
@imda-benedictlee I rebased the branch on main

**Review by imda-benedictlee — APPROVED** [review]
Approved

**Comment by berrydenhartog** [comment]
@imda-benedictlee. seems like the build fails because it can not handle forked repos
```
 RUN git clone https://github.com/imda-btg/aiverify.git --branch=fix/environmental-variables aiverify:
0.247 Cloning into 'aiverify'...
0.440 fatal: Remote branch fix/environmental-variables not found in upstream origi
```

**Comment by imda-benedictlee** [comment]
Hi @berrydenhartog, yes this has been something that I have been trying to resolve as well. Yes, if you could create a PR for this will be helpful. Thanks for the help!

**Comment by imda-benedictlee** [comment]
Hi @berrydenhartog, I realised that the secret token requirement was still in Integration Test Workflow in Main. I have updated the Integration Test Workflow in Main. Apologies, require you to pull from main again.

**Comment by berrydenhartog** [comment]
@imda-benedictlee. I rebased the PR

**Review by imda-benedictlee — APPROVED** [review]
Approved

**Comment by berrydenhartog** [comment]
@imda-benedictlee . Do you know why the CI/CD i failing? I do not understand the error so not sure what to do to fix it

**Comment by imda-benedictlee** [comment]
Pipeline Testing for Dev Script Passed
[Pipeline Testing For Dev Script.zip](https://github.com/IMDA-BTG/aiverify/files/15092515/Pipeline.Testing.For.Dev.Script.zip)

**Comment by imda-benedictlee** [comment]
Pipeline Testing for Docker Passed
[Pipeline Testing For Docker.zip](https://github.com/IMDA-BTG/aiverify/files/15095824/Pipeline.Testing.For.Docker.zip)

**Comment by imda-benedictlee** [comment]
SCA Workflow still requires some refactoring. Can ignore SCA Failure.

**Pull request merged**